### PR TITLE
Update TSP enum to union

### PR DESCRIPTION
**Pull request by lmazuel**
Swagger emitter and codegen have finished the integration of the feature that considers an union with string as an extensible enum.

This can be seen in the CADL-Ranch dashboard, test Type/Union/StringExtensibleNamed:
image

Next step is to batch update all Azure spec to the new syntax.

This should NOT change anything, as for now `enum` and `union + string` as equivalent.

For clarity, the new syntax for extensible enum is:
```typescript
// This is a extensible enum for Azure and non-Azure
union WidgetColor {
  string,
  /** The documentation for this option */
  Black: "Black",
  // If I don't like the key value I can rename it
  Green: "Teal",
  Blue: "Blue",
}
```

### CODE REVIEW

File: specification/ai/DocumentIntelligence/models.tsp
Comment (timotheeguerin) [inline specification/ai/DocumentIntelligence/models.tsp:1]: I think we also should enable the linting rule by adding

```diff
linter: 
   extends: 
     ...
+   rules:
+     "@azure-tools/typespec-azure-core/no-enum": true
```

File: specification/ai/data-plane/HealthInsights/preview/2023-09-01-preview/openapi.json [hunk 1]
```diff
@@ -1405,7 +1405,7 @@
     },
     "ClinicalTrials": {
       "type": "object",
-      "description": "The clinical trials that the patient(s) should be matched to. \nThe trial selection can be given as a list of custom clinical trials and/or a list of filters to known clinical trial registries.\nIn case both are given, the resulting trial set is a union of the two sets.",
+      "description": "The clinical trials that the patient(s) should be matched to.\nThe trial selection can be given as a list of custom clinical trials and/or a list of filters to known clinical trial registries.\nIn case both are given, the resulting trial set is a union of the two sets.",
```
Comment (mikeharder) [inline specification/ai/data-plane/HealthInsights/preview/2023-09-01-preview/openapi.json:1408]: What is causing this description change? If possible, it might be better to isolate the openapi changes into a separate PR, if the "enum to union" conversion shouldn't cause any openapi changes.

**Review by catalinaperalta — APPROVED** [review]

File: specification/ai/DocumentIntelligence/models.tsp
Comment (mikeharder) [reply specification/ai/DocumentIntelligence/models.tsp:1]: @timotheeguerin: Are you suggesting we add this to all `tspconfig.yaml` currently in the repo?  Or add this to the default ruleset specs are already required to use?  Does this rule apply to data-plane, resource-manager, or both?

File: specification/ai/HealthInsights/HealthInsights.TrialMatcher/model.trialmatcher.tsp [hunk 2]
```diff
@@ -109,7 +109,7 @@ model TrialMatcherInference {
 }
 
 @doc("""
-The clinical trials that the patient(s) should be matched to. 
+The clinical trials that the patient(s) should be matched to.
```
Comment (mikeharder) [inline specification/ai/HealthInsights/HealthInsights.TrialMatcher/model.trialmatcher.tsp:112]: I recommend moving any doc changes to a separate PR.

File: specification/ai/data-plane/OpenAI.Assistants/OpenApiV3/2024-02-15-preview/assistants_generated.yaml [hunk 3]
```diff
@@ -1419,12 +1419,14 @@ components:
           description: The files returned for the request.
       description: The response data from a file list operation.
     FilePurpose:
-      type: string
-      enum:
-        - fine-tune
-        - fine-tune-results
-        - assistants
-        - assistants_output
+      anyOf:
```
Comment (lmazuel) [inline specification/ai/data-plane/OpenAI.Assistants/OpenApiV3/2024-02-15-preview/assistants_generated.yaml:1422]: Diff on `anyOf` have been discussed and approved by @trrwilson

File: specification/ai/DocumentIntelligence/models.tsp
Comment (timotheeguerin) [reply specification/ai/DocumentIntelligence/models.tsp:1]: yeah suggesting we do that right now until the next release. So no new enum get added.  I think that's where we might want to be using a common base config with `extends` so we don't need to patch each one

File: specification/ai/HealthInsights/HealthInsights.TrialMatcher/model.trialmatcher.tsp (hunk 2 above)
Comment (lmazuel) [reply specification/ai/HealthInsights/HealthInsights.TrialMatcher/model.trialmatcher.tsp:112]: VSCode trimmed the extra space, I didn't want to reconfigure my VSCode just for a space. If it's really something we hate, I can do it though

**Comment by weidongxu-microsoft** [comment]
cadl-ranch https://github.com/Azure/cadl-ranch/pull/518

The libs also need to switch to union (from enum), this may not be a full list.
e.g.
dpg https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-core/lib/foundations.tsp#L18
mgmt https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-resource-manager/lib/models.tsp#L145

**Comment by weidongxu-microsoft** [comment]
The one concerns me as well https://github.com/Azure/azure-rest-api-specs/pull/27749/files#r1494047254

**Comment by MaryGao** [comment]
LGTM and the failure is because our RLC changelog tool didn't work yet. Issue tracked
https://github.com/Azure/azure-sdk-tools/issues/6340.

**Review by mikeharder — APPROVED** [review]
